app/assets/javascripts/views/vacation_requests_list.js

```diff
@@ -33,62 +27,64 @@ App.Views.VacationRequestsList = Backbone.View.extend({
-      if (this.isVacationVisible(model)) {
-        numberOfVisibleVacations++;
-        var item = new App.Views.VacationRequest({'model': model});
-        this.$tableRows.append(item.render().$el);
-      }
-    }, this);
-
-    if (numberOfVisibleVacations > 0) {
-      this.$table.show();
-    }
-  },
-
-  onStatusChange: function(event) {
-    this.statusFilter = this.$select.val();
-    this.renderTable();
-  },
-
-  onDateFrom: function(event) {
-    this.dateFrom = this.$from.val();
-    this.render();
+    this.$table.bootstrapTable('destroy');
+    this.$table.bootstrapTable({
+      search: true,
+      data: this.collection.toJSON(),
+      columns: [{
+          field: 'start_date',
+          title: 'Start date',
+          valign: 'middle',
+          sortable: true
+      }, {
+          field: 'actual_end_date',
+          title: 'End date',
+          valign: 'middle',
+          sortable: true
+      }, {
+          field: 'kind',
+          title: 'Type',
+          align: 'center',
+          valign: 'middle',
+          sortable: true
+      }, {
+          field: 'status',
+          title: 'Status',
+          align: 'center',
+          valign: 'middle',
+          sortable: true
+      }, {
+          field: 'operations',
+          title: 'Operations',
+          align: 'center',
+          events: this.operationsEvents(),
+          formatter: this.ownerOperationsFormatter,
+          width: '15%',
+          sortable: false
+      }],
+    });
   },
 
-  onDateTo: function(event) {
-    this.dateTo = this.$to.val();
-    this.render();
+  onCancel: function(event, value, row, index) {
+    var that = this;
+
+    $.get('vacation_requests/'+row.id.toString()+'/cancel')
+      .done(function() {
+        // TODO: implement notification, if needed
+        // Trigger table update
+        that.collection.fetch();
+      })
+      .fail(function(response) {
+        // TODO: implement notification
+        console.error('FAIL');
+      });
   },
 
-  // Convert date's string into it's miliseconds representation
-  // TODO: refactor the following code to use just Date objects
-  // as they are comparable by default.
-  dateToMS: function(date) {
-    var dateInMS = (new Date(date)).getTime();
-
-    var result = date ? dateInMS : 0;
-    return result;
-  },
-
-  // Check if given model should be visible
-  isVacationVisible: function(model) {
-    var isVisible = false;
-    // Calculate date in miliseconds to compare later
-    var dateFrom  = this.dateToMS(this.dateFrom);
-    var dateTo    = this.dateToMS(this.dateTo);
-    var modelFrom = this.dateToMS(model.get('start_date'));
-
-    var hasProperStatus = _.include(this.statusFilter, model.get('status'));
-    var hasProperDateRange = (modelFrom >= dateFrom) && ((modelFrom <= dateTo) || (dateTo === 0));
-
-    isVisible = hasProperStatus && hasProperDateRange;
+  ownerOperationsFormatter: function(value, row, index) {
+    if (row.status === 'cancelled') {
```

Comment:
Well, I thought about that. You see, I'd like to have a clue what is going on with statuses. If I am forced to store statuses in an array, I have to maintain all the changes though out the code anyway, ain't I? And, I have to memorize which status does particular index represent.

Let's see:
```JS
if (row.status === 'cancelled')
```
vs
```JS
if (row.status === App.statuses[3])
```

Which one is better?

I prefer to hardcode statuses. I just don't know how to better handle this problem in JavaScript.